### PR TITLE
Add #9789 Bookmark indentation for subfolders in bookmarks editing panel

### PR DESCRIPTION
**Pull request by junbio**
This PR adds indentation support for subfolders in the bookmarks editing panel. (#9789)

### CODE REVIEW

**Comment by junbio** [comment]
> @junbio Since this implies UI changes, could you add screenshot of what you're fixing?

Sure, no problem!

Before:
Simulator Screen Shot - iPhone 13 Pro - 2022-01-17 at 09 54 06

After:
Simulator Screen Shot - iPhone 13 Pro - 2022-01-17 at 09 50 32

**Comment by lmarceau** [comment]
@junbio Thanks! We'll get back to you shortly 😄

File: Client/Frontend/Widgets/OneLineCellAndFooter.swift [hunk 1]
```diff
@@ -17,6 +17,18 @@ enum OneLineTableViewCustomization {
 
 class OneLineTableViewCell: UITableViewCell, NotificationThemeable {
     // Tableview cell items
+    
+    override var indentationLevel: Int {
+        didSet {
+            containerView.snp.remakeConstraints { make in
+                make.height.equalTo(44)
```
Comment (lmarceau) [inline Client/Frontend/Widgets/OneLineCellAndFooter.swift:24]: Could we put the 44 in a constant with a relevant name in OneLineCellUX? We can also use it on line 74 down here.

File: Client/Frontend/Library/BookmarkDetailPanel.swift [hunk 2]
```diff
@@ -358,6 +358,7 @@ class BookmarkDetailPanel: SiteTableViewController {
                 }
 
                 cell.indentationLevel = min(item.indent, maxIndentationLevel)
+                cell.separatorInset = UIEdgeInsets(top: 0, left: CGFloat(cell.indentationLevel) * cell.indentationWidth + 61, bottom: 0, right: 0)
```
Comment (lmarceau) [inline Client/Frontend/Library/BookmarkDetailPanel.swift:361]: Same type of comment here, could we have a constant with a relevant name for this 61? So it's easier to understand context when we come back to this later on.

Comment (junbio) [reply Client/Frontend/Library/BookmarkDetailPanel.swift:361]: 61 is the current separator inset for bookmark cells without any indentation. Although it seems that only Safari indents the separator for nested bookmark folders so if it's probably a design decision.

**Comment by junbio** [comment]
> * How many level of nesting are we supporting?  What happens if there's "too much"?

There's a variable called maxIndentationLevel that gets computed and is the maximum that can be displayed. It's around 14 on an iPhone 13 Pro. 

>Should we limit the number of nesting we support?

Since Firefox on iOS syncs bookmarks with the desktop version it should be how ever many it supports.

> * We should check for dynamic type - I believe it's enabled in this section? Wanna check that this is still proper even with those changes.

Much of the indentation work had been done already; my PR just updates the UI to reflect it. Dynamic type isn't in much of the code anyways.

Thanks again for the feedback!

**Comment by lmarceau** [comment]
@junbio can you rebase? I wanna retrigger the BR

**Pull request merged**